src/python/WMCore/Services/WMStats/WMStatsPycurl.py

```diff
@@ -0,0 +1,119 @@
+"""
+This module provides some functions to create concurrent HTTP requests
+for CouchDB, as an alternative to the sequential modules available under:
+* WMCore/Services/WMStats/WMStatsReader.getTaskJobSummaryByRequest
+* WMCore/Services/WMStats/WMStatsReader.jobDetailByTasks
+
+Documentation for the parameters supported in views query can be found at:
+https://docs.couchdb.org/en/stable/api/ddoc/views.html#get--db-_design-ddoc-_view-view
+"""
+import json
+
+from urllib.parse import urljoin, urlencode
+from Utils.CertTools import ckey, cert
+from WMCore.Services.pycurl_manager import getdata as multi_getdata
+
+
+def getTaskJobSummaryByRequestPycurl(rowsSummary, sampleSize, serviceOpts):
+    """
+    Pycurl-based implementation of WMStatsReader.getTaskJobSummaryByRequest
+    :param rowsSummary: a dictionary with rows from CouchDB
+    :param sampleSize: integer with number of documents to retrieve
+    :param serviceOpts: dictionary with CouchDB options (url, db name and couchapp name)
+    :return: a dictionary with job detail
+    """
+    paramQueries = []
+    for row in rowsSummary['rows']:
+        thisQuery = {"startkey": [], "endkey": [], "numOfError": 0}
+        # row["key"] = ['workflow', 'task', 'jobstatus', 'exitCode', 'site']
+        thisQuery["startkey"] = row["key"][:4]
+        if row["key"][4]:
+            thisQuery["startkey"].append(row["key"][4])  # site
+
+        thisQuery["endkey"] = []
+        thisQuery["endkey"].extend(thisQuery["startkey"])
+        thisQuery["endkey"].append({})
+        # append amount of errors matching the 5 keys above
+        thisQuery["numOfError"] = row["value"]
+        # add query to the pool
+        paramQueries.append(thisQuery)
+
+    return jobDetailByTasksPycurl(paramQueries, sampleSize, serviceOpts)
+
+
+def jobDetailByTasksPycurl(queries, limit, serviceOpts):
+    """
+    Pycurl-based implementation of WMStatsReader.jobDetailByTasks.
+    In short, for each tuple of errors, it returns "limit" number of documents.
+    :param queries: list of CouchDB query parameters
+    :param limit: number of documents to retrieve from CouchDB
+    :param serviceOpts: dictionary with CouchDB options (url, db name and couchapp name)
+    :return: dictionary with job detail information
+
+    A decoded example of this query would be:
```

Comment:
Please either use placeholder or add explicit comment about `jobfailed` and `8006` values in queries shown below. Reader of the code may have wrong impression, e.g. look only at `jobfailed`, when in fact it may have different values. I would use `JOB_STATUS`, and `EXIT_CODE` as placeholders and then provide some values in examples below.

Reply:
Okay, to me it looks too much verbose to be added to the docstring, but here it is.

Reply:
thanks for update, it will pay off when someone will start refactoring the code. And, it show how complex the data structure is and how to address this complexity when we'll need refactoring.